### ISSUE TITLE
Pure Python wheel or WASM/Emscripten build

### ISSUE THREAD

**Issue by psychemedia** [maxbachmann/rapidfuzz#423]
I would like to be able to use `rapidfuzz` in WASM / Pyodide backed notebook environments such as JupyetrLite and marimo notebooks.

Is there any chance of getting a pure Python wheel  for `rapidfuzz`, or a WASM/Emscripten compiled wheel?

**Comment by maxbachmann** [comment]
Yes I could build both pure python wheels and wasm compiled wheels. The pure Python wheels are much smaller (somewhere around 60kb vs 777kb). However the compiled wheel should be faster (I didn't test that though).

So I think it would likely make sense to publish both versions so users can select based on their usage.
How are wheels for pyodide usually published?

**Comment by psychemedia** [comment]
In pyodide, if the PyPi wheels are universal, they can be installed using [`micropip`](https://github.com/pyodide/micropip). I think distribution of `wasm32-emscripten` wheels is still a niche thing, although [`cibuildwheel`](https://cibuildwheel.pypa.io/en/stable/options/#archs) has supported the wasm32 target for some time.

As to getting packages into pyodide, I think that is managed by requesting inclusion via an issue to the pyodide repo.

**Comment by maxbachmann** [comment]
I published a new version that properly builds for pyodide. There are already requests to add it to pyodide (https://github.com/pyodide/pyodide-recipes/issues/240) and emscripten-forge (https://github.com/emscripten-forge/recipes/issues/1270). So hopefully it will get added there soon.

In the meantime you can use the attached wheel and just host it yourself.

[pyodide-wheel.zip](https://github.com/user-attachments/files/18168509/pyodide-wheel.zip)

**Comment by psychemedia** [comment]
Fantastic, thanks.

**Comment by psychemedia** [comment]
From a quick upload test to [JupyterLite](https://jupyterlite.rtfd.io/en/stable/try/lab), neither the zip nor the wheel can be imported using `micropip.install()`. I'm not sure if this is a wheel name convention thing or something other.

**Comment by maxbachmann** [comment]
I don't know much about the pyodide ecosystem. I used the attached test environment. There I did run:
- `python cors_server.py` to make the wheel available with a working access control
- opened the `index.html` in a browser

This did work for me:
![image](https://github.com/user-attachments/assets/becb86e6-861c-42f1-b452-2faf70d5ff13)


[testEnv.zip](https://github.com/user-attachments/files/18172979/testEnv.zip)

**Comment by agriyakhetarpal** [comment]
> From a quick upload test to [JupyterLite](https://jupyterlite.rtfd.io/en/stable/try/lab), neither the zip nor the wheel can be imported using `micropip.install()`. I'm not sure if this is a wheel name convention thing or something other.

Linked issue: https://github.com/pyodide/micropip/issues/158

**Comment by agriyakhetarpal** [comment]
> From a quick upload test to [JupyterLite](https://jupyterlite.rtfd.io/en/stable/try/lab), neither the zip nor the wheel can be imported using `micropip.install()`. I'm not sure if this is a wheel name convention thing or something other.

This was probably because we changed the ABI tag from `emscripten_X_Y_ZW` to `pyodide_YYYY_N`. Linked issue: https://github.com/pyodide/micropip/issues/158

**Comment by maxbachmann** [comment]
@agriyakhetarpal if I understand this correctly this is fixed in micropip and I can close this here right?

**Comment by agriyakhetarpal** [comment]
Yes, @maxbachmann – the fix was released with micropip 0.7.2, and jupyterlite-pyodide-kernel's versions 0.4.7 and later should be able to install wheels with both ABI tags.

**Comment by psychemedia** [comment]
If you unzip the `pyodide-wheel.zip` and upload the `.whl` file, then it can be installed and test in jupyterlite as follows:

```python
# Install package; files in JupyterLite are uploaded to and exist on the path /drive
import micropip
await micropip.install("emfs:/drive/rapidfuzz-3.11.0-cp312-cp312-pyodide_2024_0_wasm32.whl")
```

<img width="859" alt="image" src="https://github.com/user-attachments/assets/fa3dbe1f-3b2a-4d74-881e-4f557789d05c" />

The package can then be imported and run as you might expect:

```python
from rapidfuzz import fuzz
fuzz.ratio("this is a test", "this is a test!")

>>> 96.55172413793103
```
